### ISSUE TITLE
fdisk not working

### ISSUE THREAD

**Issue by TanMan1217** [Microsoft/BashOnWindows#837]
Running fdisk -l in bash on 14393.10 as administrator. Error is "cannot open /proc/partitions". cat /proc/partitions gets error "No such file or directory". It appears that /proc/partitions is not yet implemented.

**Comment by Pachonk** [comment]
+1

**Comment by fpqc** [comment]
I'm going to go out on a limb here and say that support for fdisk is not planned and not likely over the course of the next full calendar year.

**Comment by Pachonk** [comment]
Agreed. I was hoping to get fdisk and dd working to be able to dd, since any Windows dd doesn't work very well.

**Comment by russalex** [comment]
Thanks for the feedback.

As @fpqc points out, tools for managing your drive partitions are not supported and are very low on our priority list.  As always, to make certain the ask is kept I would recommend heading to our [User Voice](https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo/category/161892-bash) page and describing a bit about what you're trying to do.  While we won't be addressing this one in the near future, it will help us understand the priority longer term.

**Comment by francesco1119** [comment]
@fpqc fdisk not supported. @russalex "...are very low on our priority list".
So why Bash was ported to Windows 10 for? 
Is there a list of the command that **are** supported?

**Comment by aseering** [comment]
@francesco1119 -- the thing that's really not supported right now is direct access to physical hardware.  You're welcome to run fdisk all you want, but it can't actually read your disk so there's not much it can do...

There's a community-maintained of known-working (and known-not-working) applications here:
http://ethanhs.me/WSL-Programs/

A lot of people who I've personally heard using WSL are developing network services or Web apps that they are going to deploy on Linux servers.

On an official level, the WSL team is not porting applications; they are implementing kernel functionality.  You can try to run any existing Linux binary; it will work if all of the functionality it needs is available.  If you want low-level details, you can find them from here:
https://msdn.microsoft.com/en-us/commandline/wsl/release_notes

**Comment by SystemDisc** [comment]
For anyone who wants to voice their support for this, you can vote for this on User Voice: https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo/suggestions/19234984-support-tools-like-fdisk-and-dd-to-directly-accss

**Comment by fpqc** [comment]
@SystemDisc https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo/suggestions/13396686-access-to-real-linux-devices-in-dev this one has more votes and is the same idea

**Comment by SystemDisc** [comment]
Oh, thank you! I did I search but didn't find that one.

**Comment by samProuse** [comment]
Support for physical drives is (whether or not it was) achievable, but DD still won't work, since you have to mount it to a /mnt/(letter) directory.

**Comment by fpqc** [comment]
@samProuse No no. DrvFS is not done using direct hardware access. It goes through the Windows HAL.  There are no /dev block devices.

**Comment by DarthSpock** [comment]
@fpqc That being said, I did see a BlueHat presentation from MS showing how WSL can reach into the NT kernel. Of course it was using shm for the purposes of an exploit that's already been addressed (which makes me wonder why they used an older build when it was posted this month on Youtube). But that does prove access to the kernel can be forced through and might allow someone _**clever**_ to access the hardware. Might be a bit _hacky_ but for now it would be better than nothing as it could allow for some exciting possibilities.

**Comment by xiuyuanjun** [comment]
+1
